### ISSUE TITLE
Support for projects unlinked from root

### ISSUE THREAD

**Issue by blast-hardcheese** [scala-steward-org/scala-steward#1481]
For some time, I was wondering why only some libraries on https://github.com/twilio/guardrail/ were receiving library update PRs.

After digging in a bit, running `stewardDependencies` locally, I discovered that due to `stewardDependencies` collecting `libraryDependencies` from the project it was launched in, if there were any projects that were only accessible fully qualified, they'd be completely left out of analysis.

To put it differently,
```
sbt> stewardDependencies // Only lists a couple dependencies, defined on `root`
sbt> sampleAkkaHttp/stewardDependencies // Finds dependencies that need to be updated
```

The reason for this is surely due to my lack of how to properly structure an SBT project to serve my needs, but the guardrail integration tests are run in subprojects that are only able to be `compile`d during integration tests, as all other times the required source files have not been generated yet.

I was able to play a little shell game with my dependencies, which works pretty decently: https://github.com/twilio/guardrail/pull/664

... though I wonder if there's not a more elegant solution for this problem, perhaps `.steward.conf` containing `projects = foo, bar, baz` or similar.

This seems somewhat related to https://github.com/fthomas/scala-steward/issues/1158 as well, though that issue seems to be with multiple, separate invocation roots, whereas mine is just separate command prefixes plus an aggregation phase.

As always, thank you for this excellent tool, and have a fantastic day writing Scala!

**Comment by fthomas** [comment]
You're right that Scala Steward only updates dependencies that are listed by `libraryDependencies` of the root project. Running that task in the root project will only run it in subprojects that are aggregated by the root project. I'm not aware of a way to list dependencies for all subprojects regardless whether they are aggregated by the root project or not. If it exists I'm open to discussing if Scala Steward should be changed in this direction.

I also think that https://github.com/twilio/guardrail/pull/664 looks like a good workaround when you don't want certain subprojects aggregated in the root project but still want to benefit from Scala Steward's automatic updates. The good thing about this solution is that library definitions are still only in the sbt build and do not need to be duplicated in some Scala Steward specific config file.

**Comment by eugeniyk** [comment]
@blast-hardcheese what we endup with giter8 templates (same - empty root dependencies, but lot's inside) is to regenerate dependencies.gen.sbt inside root that will aggregate them (simple bash scripts ;)

Then Scala steward update in both places - root (gen) and under

**Comment by exoego** [comment]
IIUC, this is covered by repo-specific config
```
buildRoots = [ ".", "subfolder/projectA" ]
```
which was introduced in #1875
So let me close as resolved.
See https://github.com/scala-steward-org/scala-steward/blob/main/docs/repo-specific-configuration.md for config detail.

**Comment by satorg** [comment]
@exoego I don't think that `buildRoots` feature resolves the original concern.

`buildRoots` handles multiple independent projects defined in their own `build.sbt` files within the same mono repository.

Whereas the former concern was about projects that are defined in the same `build.sbt` but not aggregated into the root.
```scala
lazy val first = project
  .settings(
    libraryDependencies ++= Seq(...)
  )

lazy val second = project
  .settings(
    libraryDependencies ++= Seq(...)
  )

// This project is not aggregated into the root!
lazy val other = project
  .settings(
    // These dependencies are not checked/updated by
    // scala-steward!
    libraryDependencies ++= Seq(...)
  )

lazy val root = project.in(file("."))
  .aggregate(first, second) // but not `other`!
```
That one doesn't seem to be addressed yet.

cc @mzuehlke

**Comment by satorg** [comment]
FYI: #3818
